### PR TITLE
[Storage] Fix #10620 in azure cli to support ADLS

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

File: src/storage-preview/azext_storage_preview/tests/latest/test_storage_adls_scenarios.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        self.storage_cmd('storage blob directory access set -c {} -d {} -a "{}"', account_info, container, des_directory, acl)
+        self.storage_cmd('storage blob directory access show -c {} -d {}', account_info, container, des_directory) \
+            .assert_with_checks(JMESPathCheck('acl', acl))
+        self.storage_cmd('storage blob directory access update -c {} -d {} --permissions "rwxrwxrwx"', account_info,
+                         container, des_directory, acl)
+        self.storage_cmd('storage blob directory access show -c {} -d {}', account_info, container,
+                         des_directory).assert_with_checks(JMESPathCheck('permissions', "rwxrwxrwx"))
+
+        # Storage blob directory metadata
+<<<<<<< HEAD
```
Comment (zikalino) [inline src/storage-preview/azext_storage_preview/tests/latest/test_storage_adls_scenarios.py]: seems like merge mistake

File: src/storage-preview/azext_storage_preview/_transformers.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+
+def transform_storage_list_output(result):
+    if getattr(result, 'next_marker', None):
+        logger.warning('Next Marker:')
+        logger.warning(result.next_marker)
+    return list(result)
+
+
+def transform_url(result):
```
Comment (zikalino) [inline src/storage-preview/azext_storage_preview/_transformers.py]: seems this is not used anywhere, and if it was why would url contain extra characters?

File: src/storage-preview/azext_storage_preview/operations/blob.py [hunk 3]
```diff
@@ -36,3 +43,205 @@ def set_service_properties(client, parameters, delete_retention=None, days_retai
...
+
+    return blob
+
+
+def guess_content_type(file_path, original, settings_class):
+    if original.content_encoding or original.content_type:
+        return original
+
+    import mimetypes
+    mimetypes.add_type('application/json', '.json')
```
Comment (zikalino) [inline src/storage-preview/azext_storage_preview/operations/blob.py]: why adding these types specifically?

Comment (Juliehzl) [reply src/storage-preview/azext_storage_preview/operations/blob.py]: removed

**Pull request merged**